hw/drivers/flash/spiflash/src/spiflash.c

```diff
@@ -748,6 +794,8 @@ spiflash_read(const struct hal_flash *hal_flash_dev, uint32_t addr, void *buf,
 
     dev = (struct spiflash_dev *)hal_flash_dev;
 
+    spiflash_lock(dev);
+
     err = spiflash_wait_ready(dev, 100);
```

Comment:
Just do be sure, you are locking just above and again inside `spiflash_wait_ready`. I never remember the semantics of these mutexes but does this not cause a deadlock?

Reply:
```
Note that nested calls to`os_mutex_pend` are allowed but there needs to be one release per pend.
```

Answering my own question! :-P